eitprocessing/features/pixel_inflation.py

```diff
@@ -0,0 +1,173 @@
+            An IntervalData object containing Breath objects.
+        """
+        if store is None and isinstance(sequence, Sequence):
+            store = True
+
+        if store and sequence is None:
+            msg = "Can't store the result if no Sequence is provided."
+            raise RuntimeError(msg)
+
+        if store and not isinstance(sequence, Sequence):
+            msg = "To store the result a Sequence dataclass must be provided."
+            raise ValueError(msg)
+
+        bd_kwargs = self.breath_detection_kwargs.copy()
+        breath_detection = BreathDetection(**bd_kwargs)
+        breaths = breath_detection.find_breaths(continuous_data)
+
+        indices_breath_middles = np.searchsorted(eit_data.time, [breath.middle_time for breath in breaths.values])
+
+        _, rows, cols = eit_data.pixel_impedance.shape
+
+        from eitprocessing.parameters.tidal_impedance_variation import TIV
+
+        tiv_result_pixel_inspiratory_global_timing = TIV().compute_pixel_parameter(
+            eit_data,
+            continuous_data,
+            sequence,
+            tiv_method="inspiratory",
+            tiv_timing="continuous",
+        )
+
+        mean_tiv_pixel = np.nanmean(tiv_result_pixel_inspiratory_global_timing, axis=0)
+        time = eit_data.time
+        pixel_impedance = eit_data.pixel_impedance
+
+        pixel_inflations = np.full((len(breaths), rows, cols), None)
+
+        for row in range(rows):
+            for col in range(cols):
+                mean_value = mean_tiv_pixel[row, col]
+
+                if mean_value == 0.0:
+                    continue
+
+                if mean_value < 0:
+                    mode_start, mode_middle = np.argmax, np.argmin
+                else:
+                    mode_start, mode_middle = np.argmin, np.argmax
+
+                start = self._find_extreme_indices(pixel_impedance, indices_breath_middles, row, col, mode_start)
+                end = start[1:]
+                middle = self._find_extreme_indices(pixel_impedance, start, row, col, mode_middle)
+
+                ## To discuss: this block of code is implemented to prevent noisy pixels from breaking the code.
+                # Quick solve is to make entire breath object None if any breath in a pixel does not have
+                # consecutive start, middle and end.
+                # However, this might cause problems elsewhere.
+
+                if (start[:-1] >= middle).any() or (middle >= end).any():
+                    inflations = None
+                else:
+                    start = start[:-1]
+                    inflations = self._compute_inflations(start, middle, end, time)
+                pixel_inflations[:, row, col] = inflations
+
+        pixel_inflations_container = IntervalData(
+            label=result_label,
+            name="Pixel in- and deflation timing as determined by PixelInflation",
+            unit=None,
+            category="breath",
+            intervals=[
```

Comment:
I agree that this does not seem logical. The intervals now run from middle to middle of the continuous data breath which is the interval in which we detect the start of the pixel inflation. It might make more sense to just use the start and end time of the pixel inflation as interval? To me it does make sense to store it as intervaldata just as breaths. But maybe you have a different suggestion?

Reply:
I would certainly not store it as the interval from middle to middle. If it is stored as intervaldata, I would store it with the same intervals of the continuous data Breaths each inflation is associated with. In that way, when you slice, you actually keep the pixel inflations that belong to all breaths that you select.

If we store them without any timing data, they lose their slice ability, which means you have to filter them yourself. 

For now, let's keep it as IntervalData, but set the interval as the start/end of the breath the pixel inflation is associated with. If it requires something different, it's probably a new type of storage (which I'm not keen on introducing).

Reply:
If the interval is set from start to end of a normal breath and the pixel inflation is slightly delayed, wouldn't that make you miss the end of the pixel inflation in the interval?

Reply:
Yes. However, is this a problem? If I want to select 10 breaths, I want 10 inflations per pixel. If a pixel is early or late, and you include everything that overlaps even 1 timeframe, you get 11 inflations for many/most pixels. This does not match the number of breaths.

I think for now this is the best we can do. Let's see how it works when using it, and whether we run into issues.